### PR TITLE
dovecot: avoid testing DES-encrypted passwords

### PR DESCRIPTION
**Pull request by vcunat**
Those tests wouldn't work since #220557

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
Dovecot maintainers should confirm that dropping DES support here shouldn't cause trouble in practice: @fpletz, @globin, @ajs124.

**Comment by vcunat** [comment]
Perhaps you'd like to also like to add `dovecot` to the list of packages explicitly mentioned in [release notes](https://github.com/NixOS/nixpkgs/pull/220557/files)?

**Comment by ajs124** [comment]
> Perhaps you'd like to also like to add `dovecot` to the list of packages explicitly mentioned in [release notes](https://github.com/NixOS/nixpkgs/pull/220557/files)?

Sounds like a good idea.

**Review by fpletz — APPROVED** [review]
When using a custom passwd/shadow file for dovecot, the command `doveadm pw` generates a bcrypt based hash by default - even before the libxcrypt changes. I don't think anyone still uses DES and if they do, they should upgrade anyways (e.g. because only the first 8 characters of the plain password are actually hashed).

**Comment by vcunat** [comment]
IIRC the default changed at the start of `dovecot-2.3`, so something like 2017.

**Review by ajs124 — APPROVED** [review]

File: pkgs/servers/mail/dovecot/default.nix [hunk 1]
```diff
@@ -41,6 +41,9 @@ stdenv.mkDerivation rec {
     sed -i -s -E 's!\bcat\b!${coreutils}/bin/cat!g' src/lib-smtp/test-bin/*.sh
 
     patchShebangs src/config/settings-get.pl
+
+    # DES-encrypted passwords are not supported by NixPkgs anymore
```
Comment (SuperSandro2000) [inline pkgs/servers/mail/dovecot/default.nix:45]: nit Nixpkgs

**Pull request merged**